ksiren/src/main/java/com/brightspace/ksiren/Action.kt

```diff
@@ -86,14 +87,24 @@ data class Action(
 		}
 	}
 
-	fun hasField(name: String): Boolean {
-		this.fields.forEach {
-			if (it.name == name) {
-				return true
+	fun deepCopy(): Action = copy(fields = fields.map { field -> field.copy() })
+
+	fun hasField(name: String): Boolean = this.fields.any { field -> field.name == name }
+
+	override fun toJson(writer: KSirenJsonWriter): String = JsonUtils.toJson(this, writer)
+
+	fun toJsonRequestBody(writer: KSirenJsonWriter): String {
+		writer.use {
+			writer.beginObject()
+
+			fields.forEach { field ->
+				writer.name(field.name)
+				writer.value(field.value)
 			}
+
+			writer.endObject()
 		}
-		return false
-	}
 
-	override fun toJson() = JsonUtils.toJson(this)
+		return writer.getSerializedString()
+	}
```

Comment:
nit: This _could_ be a little...less:

```
fun toJsonRequestBody(writer: KSirenJsonWriter) = writer.use {
	with(it) {
		beginObject()
		fields.forEach { name(it.name).value(it.value) }
		endObject().getSerializedString()
	}
}
```

Or, if you need to close before calling `getSerializedString()`:

```
fun toJsonRequestBody(writer: KSirenJsonWriter) = writer.use {
	with(it) {
		beginObject()
		fields.forEach { name(it.name).value(it.value) }
		endObject()
	}
}.getSerializedString()
```